src/twisted/words/im/basesupport.py

```diff
@@ -127,7 +130,7 @@ def connectionMade(self):
     def connectionLost(self, reason: Failure = connectionDone):
         self.account._clientLost(self, reason)
         self.unregisterAsAccountClient()
-        return self._protoBase.connectionLost(self, reason)
+        return self._protoBase.connectionLost(self, reason)  # type: ignore[arg-type]  # noqa
```

Comment:
why is this required?

Reply:
Without that `ignore`, this error is returned:

```
src/twisted/words/im/basesupport.py:133:47: error: Argument 1 to "connectionLost" of "Protocol" has incompatible type "AbstractClientMixin"; expected "Protocol"  [arg-type]
```

Reply:
Thanks. I think that is ok to comment it.

AbstractClientMixin is not really a mixing ... it has an init is part of multiple inheritance.
I am expecting from a mixing to not have any `__init__` and to always inherit it last